File: src/main/java/template_method/PostAccountController.java
package template_method;

public class PostAccountController extends BasePostController<PostAccountReqBody, PostAccountResBody, String> {

    public Response<?> createAccount(PostAccountReqBody requestBody) {
        return execute(requestBody);
    }

    @Override
    protected String mapToDto(PostAccountReqBody requestBody) {
        return requestBody.branchCode() + requestBody.accountNumber();
    }

    @Override
    protected PostAccountResBody doService(String accountNumber) {
        return new PostAccountResBody(accountNumber);
    }

    @Override
    protected Response<PostAccountResBody> createResponse(PostAccountResBody responseBody) {
        return new Response<>(responseBody, 201);
    }
}

File: src/main/kotlin/template_method/PostAccountController.kt
package template_method

class PostAccountController : BasePostController<PostAccountReqBody, PostAccountResBody, String>() {
    fun createAccount(requestBody: PostAccountReqBody): Response<*> {
        return execute(requestBody)
    }

    override fun mapToDto(requestBody: PostAccountReqBody): String {
        return requestBody.branchCode + requestBody.accountNumber
    }

    override fun doService(accountNumber: String): PostAccountResBody {
        return PostAccountResBody(accountNumber)
    }

    override fun createResponse(responseBody: PostAccountResBody): Response<PostAccountResBody> {
        return Response(responseBody, 201)
    }
}